google/resource_cloudfunctions_function.go

```diff
@@ -550,9 +556,30 @@ func flattenEventTrigger(eventTrigger *cloudfunctions.EventTrigger) []map[string
 		return result
 	}
 
+	resource := ""
+	switch {
+	case strings.HasPrefix(eventTrigger.EventType, "google.storage.object."):
+		resource = GetResourceNameFromSelfLink(eventTrigger.Resource)
+	case strings.HasPrefix(eventTrigger.EventType, "google.pubsub.topic."):
+		resource = GetResourceNameFromSelfLink(eventTrigger.Resource)
+	case strings.HasPrefix(eventTrigger.EventType, "google.firestore.document."):
+		parts := strings.SplitN(eventTrigger.Resource, "/", 6)
+		resource = parts[len(parts)-1]
+		// Legacy style triggers
+	case strings.HasPrefix(eventTrigger.EventType, "providers/cloud.storage/eventTypes/"):
+		resource = GetResourceNameFromSelfLink(eventTrigger.Resource)
+	case strings.HasPrefix(eventTrigger.EventType, "providers/cloud.pubsub/eventTypes/"):
+		resource = GetResourceNameFromSelfLink(eventTrigger.Resource)
+	case strings.HasPrefix(eventTrigger.EventType, "providers/cloud.firestore/eventTypes/"):
+		parts := strings.SplitN(eventTrigger.Resource, "/", 6)
```

Comment:
I'm assuming the `6` is significant here, and we can't just use `GetResourceNameFromSelfLink` because we need to store a value like `messages/{messageId}`. Is that correct?

If you don't mind, could you comment including an example of a full `parts` value and explain why we need to deviate here?

Reply:
I added a comment detailing the specifics of this